VM & VMSS MSI rework and support for "latest" version parameter in GetImage

anuchandy opened the pull request:
Addresses
    [1]. https://github.com/Azure/azure-sdk-for-java/issues/1844
    [2]. https://github.com/Azure/azure-sdk-for-java/issues/1392

martinsawicki commented on azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java:
```diff
@@ -1501,48 +1501,73 @@ WithManagedCreate withNewDataDiskFromImage(int imageLun,
             /**
-             * Specifies that Managed Service Identity needs to be enabled in the virtual machine.
+             * Specifies that applications running on the virtual machine requires the given access role
+             * with scope of access limited to the arm resource identified by the resource id specified
```
"arm" is an acronym (ARM)

martinsawicki replied:
"id" is capitalized as "ID" in user-facing text

martinsawicki commented on azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java:
```diff
@@ -1501,48 +1501,73 @@ WithManagedCreate withNewDataDiskFromImage(int imageLun,
             /**
-             * Specifies that Managed Service Identity needs to be enabled in the virtual machine.
+             * Specifies that applications running on the virtual machine requires the access described
+             * in the given role definition with scope of access limited to the arm resource identified
```
ARM

anuchandy commented:
Vlad showed that the test i skipped in fact works, propably it didn't work for me due to not copying the recording correctly. Just removed the test from ignored list and updated the recording.

anuchandy commented:
@martinsawicki @milismsft @lenala @hovsepm @alvadb @jianghaolu @vladimir-shcherbakov @RikkiGibson  FYI if you have a scenario where you wanted to create an observable from a map and eventually flatmap it for parallelization, make sure the concrete type of map is TreeMap or LinkedHashMap (i.e. the DS that honor ordering). Don't use HashMap (which don't have ordering). The reason is - in playback mode, the request-response pairs are evaluated in order, which is by design.

martinsawicki commented:
@anuchandy quick question -- there are no compat breaks with any previous GA or Beta functionality, right?

anuchandy commented:
@martinsawicki  no, addition of MSI is not a breaking change.
